apps/workspace-host/src/interface.js

```diff
@@ -490,13 +490,17 @@ async function _allocateContainerPort(workspace) {
  * @param {function} callback
  */
 function _checkServer(workspace, callback) {
-  const checkMilliseconds = 500;
-  const maxMilliseconds = 30000;
+  const startTimeout = config.workspaceStartTimeoutSec * 1000;
+  const healthCheckInterval = config.workspaceHealthCheckIntervalSec * 1000;
+  const healthCheckTimeout = config.workspaceHealthCheckTimeoutSec * 1000;
 
   const startTime = new Date().getTime();
   function checkWorkspace() {
     request(
-      `http://${workspace_server_settings.server_to_container_hostname}:${workspace.launch_port}/`,
+      {
+        url: `http://${workspace_server_settings.server_to_container_hostname}:${workspace.launch_port}/`,
+        timeout: healthCheckTimeout,
```

Comment:
This is new. I figured we should have been doing this all along. This keeps a workspace from accepting a connection but never closing it.